src/encodings/encoding.rs

```diff
@@ -630,6 +636,134 @@ impl DeltaBitPackEncoderConversion<Int64Type> for DeltaBitPackEncoder<Int64Type>
+    self.len_encoder.put(&lengths)?;
+    for byte_array in values {
+      self.data.extend_from_slice(byte_array.data());
+    }
+    Ok(())
+  }
+
+  fn flush_buffer(&mut self) -> Result<ByteBufferPtr> {
+    let mut total_bytes = vec!();
+    let lengths = self.len_encoder.flush_buffer()?;
+    total_bytes.extend_from_slice(lengths.data());
+    total_bytes.extend_from_slice(&self.data);
+    self.data.clear();
+    Ok(ByteBufferPtr::new(total_bytes))
+  }
+}
+
+/// Encoding for byte arrays, prefix lengths are encoded using DELTA_BINARY_PACKED
+/// encoding, followed by suffixes with DELTA_LENGTH_BYTE_ARRAY encoding.
+pub struct DeltaByteArrayEncoder<T: DataType> {
+  prefix_len_encoder: DeltaBitPackEncoder<Int32Type>,
+  suffix_writer: DeltaLengthByteArrayEncoder<T>,
+  previous: Vec<u8>,
+  _phantom: PhantomData<T>
+}
+
+impl<T: DataType> DeltaByteArrayEncoder<T> {
+  pub fn new() -> Self {
+    Self {
+      prefix_len_encoder: DeltaBitPackEncoder::<Int32Type>::new(),
+      suffix_writer: DeltaLengthByteArrayEncoder::<T>::new(),
+      previous: vec!(),
+      _phantom: PhantomData
+    }
+  }
+}
+
+default impl<T: DataType> Encoder<T> for DeltaByteArrayEncoder<T> {
+  fn put(&mut self, _values: &[T::T]) -> Result<()> {
+    panic!("DeltaByteArrayEncoder only supports ByteArrayType");
+  }
+
+  fn encoding(&self) -> Encoding {
+    Encoding::DELTA_BYTE_ARRAY
+  }
+
+  fn flush_buffer(&mut self) -> Result<ByteBufferPtr> {
+    panic!("DeltaByteArrayEncoder only supports ByteArrayType");
+  }
+}
+
+impl Encoder<ByteArrayType> for DeltaByteArrayEncoder<ByteArrayType> {
+  fn put(&mut self, values: &[ByteArray]) -> Result<()> {
+    let mut prefix_lengths: Vec<i32> = vec!();
+    let mut suffixes: Vec<ByteArray> = vec!();
+
+    for byte_array in values {
+      let current = byte_array.data();
+      // Maximum prefix length that is shared between previous value and current value
+      let prefix_len = cmp::min(self.previous.len(), current.len());
+      let mut match_len = 0;
+      while match_len < prefix_len && self.previous[match_len] == current[match_len] {
+        match_len += 1;
+      }
+      prefix_lengths.push(match_len as i32);
+      suffixes.push(byte_array.slice(match_len, byte_array.len() - match_len));
+      // Update previous for the next prefix
+      self.previous.clear();
+      self.previous.extend_from_slice(current);
+    }
+    self.prefix_len_encoder.put(&prefix_lengths)?;
+    self.suffix_writer.put(&suffixes)?;
+    Ok(())
+  }
+
+  fn flush_buffer(&mut self) -> Result<ByteBufferPtr> {
+    let mut total_bytes = vec!();
+    // Insert lengths ...
+    let lengths = self.prefix_len_encoder.flush_buffer()?;
+    total_bytes.extend_from_slice(lengths.data());
+    // ... followed by suffixes
+    let suffixes = self.suffix_writer.flush_buffer()?;
+    total_bytes.extend_from_slice(suffixes.data());
+    // Reset previous value
+    self.previous.clear();
```

Comment:
Probably not. I added it, because I thought if you reuse encoder after `flush_buffer` then we should reset previous value. But it looks like we should not. Will remove.